dev-tools/docker/linux_image/Dockerfile

```diff
@@ -161,11 +164,42 @@ RUN \
   export USE_XNNPACK=OFF && \
   export PYTORCH_BUILD_VERSION=1.13.1 && \
   export PYTORCH_BUILD_NUMBER=1 && \
-  /usr/local/bin/python3.10 setup.py install && \
+  export MAX_JOBS=10 && \
+  /usr/local/bin/python3.10 setup.py install
+
+RUN \
+  cd ${build_dir} && \
+  git clone https://github.com/intel/intel-extension-for-pytorch.git && \
+  cd intel-extension-for-pytorch && \
+  git fetch --all && \
+  git checkout v1.13.100+cpu && \
+  git submodule sync && \
+  git submodule update --init --recursive && \
+  git config --global --add safe.directory `pwd` && \
+  /usr/local/bin/python3.10 -m pip install -r requirements.txt && \
+  /usr/local/bin/python3.10 setup.py clean
+
+RUN  \
+  cd ${build_dir} && \
+  echo "1.13.1+cpu\n" > ${build_dir}/pytorch/torch/build-version
+
+RUN \
+  cd ${build_dir}/intel-extension-for-pytorch && \
+  export  CC=/usr/local/gcc103/bin/gcc  && \
+  export TORCH_VERSION="v1.13.1" && \
+  export TORCH_IPEX_VERSION="1.13.100+cpu" && \
+  # building ipex is memory hungry, so we need to limit the number of jobs
+  sed -i 's/multiprocessing.cpu_count()/1/g' setup.py && \
+  /usr/local/bin/python3.10 setup.py build_clib /usr/src/pytorch/torch
+
+
+RUN \
+  cd ${build_dir}/pytorch && \
   mkdir /usr/local/gcc103/include/pytorch && \
   cp -r torch/include/* /usr/local/gcc103/include/pytorch/ && \
   cp torch/lib/libtorch_cpu.so /usr/local/gcc103/lib && \
   cp torch/lib/libc10.so /usr/local/gcc103/lib && \
+  cp /usr/src/intel-extension-for-pytorch/build/Release/packages/intel_extension_for_pytorch/lib/libintel-ext-pt-cpu.so /usr/local/gcc103/lib && \
   cd .. && \
   rm -rf pytorch
```

Comment:
You've split the building of PyTorch over multiple `RUN` steps. The reason it was all in one before was that trying to delete the `pytorch` directory after having built it up via multiple file system layers sometimes caused errors when we first tried that. Maybe it's OK with certain versions of Docker but not others.

The other consideration with using multiple `RUN` steps is the disk space needed by Docker. The build script warns if less than 50GB is available, but does it now need to say that more is required?

Reply:
Indeed, thank you for the hint. As this was a WIP I needed to be able to skip some more expensive steps repeating the same process multiple times. I squashed all the steps for torch and ipex in one RUN.